### PR TITLE
chore(clippy): clear Rust 1.97 warnings

### PR DESCRIPTION
**Pull request by Audacity88**
## Summary

- **Base branch:** `master`
- **What changed and why:** Applied six behavior-preserving Clippy suggestions across the tool-call parser, runtime, and tools crates: simplified two `Option` early-return branches, removed redundant wildcard destructuring, removed an unnecessary formatting borrow, and removed two redundant references passed to `write!`.
- **Scope boundary:** This PR does not change runtime behavior, public APIs, tests, dependencies, or feature wiring.
- **Blast radius:** Compile-time lint cleanliness for `zeroclaw-tool-call-parser`, `zeroclaw-runtime`, and `zeroclaw-tools`.
- **Linked issue(s):** Related #9527.
- **Labels:** `runtime`, `skills`, `tool`, `distinguished contributor`, `type:refactor`, `risk:low`, `size:XS`

### How you can test (when useful)

- **Reviewer testing requested?** N/A; these are mechanical compiler-suggested rewrites with no user-visible behavior change.

### How I tested

- **CI checks relied on and why they cover this change:** Required GitHub CI completed on the current head. Security fails because `cargo audit` flags newly published `RUSTSEC-2026-0219` in unchanged `nostr 0.44.5`; this PR changes no dependencies. Parallel Runtime Test failed in an unchanged `agent::loop_` lifecycle test after its global event recorder observed another test's events under parallel execution; none of this PR's runtime hunks touch that recorder or test. CI Required Gate is red as the aggregate result of those failures; every other check passes. Final green status remains a prerequisite for maintainer sign-off. Focused exact-head Clippy directly checks all three affected library crates with warnings denied.
- **Known CI coverage gap, if any:** The focused command checks library targets rather than the entire workspace; every modified expression is in a library target.
- **Commands run and tail output:**

```text
cargo clippy \
  -p zeroclaw-tool-call-parser \
  -p zeroclaw-runtime \
  --lib --no-deps -- -D warnings
Finished `dev` profile [unoptimized + debuginfo] target(s) in 2m 24s

cargo clippy \
  -p zeroclaw-tools \
  --lib --no-deps -- -D warnings
Finished `dev` profile [unoptimized + debuginfo] target(s) in 3m 01s

git diff --check
# passed

bash scripts/ci/comment_hygiene_gate.sh
Comment hygiene gate passed.
```

- **Beyond CI, what did you manually verify?** Compared each before/after expression and confirmed that return behavior, matched values, formatting output, and event serialization are unchanged.
- **If any command was intentionally skipped, why:** Workspace-wide Clippy was not run because the reported warnings and all modified expressions are covered by the focused three-package library checks.

## Security & Privacy Impact (required)

- New permissions, capabilities, or file system access scope? No
- New external network calls? No
- Secrets / tokens / credentials handling changed? No
- PII, real identities, or personal data in diff, tests, fixtures, or docs? No
- Prompt injection or untrusted model-visible text introduced/changed? No
- If any Yes, describe the risk and mitigation: N/A

## Compatibility (required)

- Backward compatible? Yes
- Config / env / CLI surface changed? No
- Rust/MSRV/toolchain floor changed? No
- If backward compatibility is No or either surface/floor question is Yes: N/A

## Rollback (required for medium/high-risk PRs)

Low risk; revert the eventual merge commit.

### CODE REVIEW

**Review by IftekharUddin — APPROVED** [review]
@Audacity88, I reviewed exact head 2bb76b9183ad5f861ad5d610abd191c0c2112335 against master 04af7b83868d86fb7ad9fd8ee671524aa1f725cc.

### 🟢 What looks good — the lint rewrites preserve all four behavior boundaries

The `TurnEvent::Usage` match still binds the same `input_tokens` value and ignores the same remaining fields. The `truncate_output` formatter still formats the same owned replacement string. The SOP lookup still returns `None` for an absent key while holding the same read lock, and the GLM parser still returns `None` when no `>` exists. No new state, dependency, public API, configuration, permission, or user-visible text is introduced.

### 🟢 What looks good — the evidence matches the narrow scope

The current-head CI set is terminal with 29 successful checks and three intentional skips. On the clean synthetic merge, focused Clippy passed for both affected library crates; the parser tests passed 6/6, runtime truncation tests passed 2/2, runtime usage-event tests passed 5/5, and SOP metrics tests passed 26/26. Formatting and diff hygiene also passed, and the synthetic merge tree matched Git's independent `merge-tree` result.

The PR body, labels, `v0.8.5` milestone, compatibility statement, security/privacy statement, and low-risk rollback plan match the actual diff. I found no blocking, warning, or suggestion findings.

**Comment by NiuBlibing** [comment]
While bumping the CI toolchain to Rust 1.97.1 (#9527), the targeted Windows Clippy job surfaced two more `clippy::question_mark`-adjacent lints that this PR doesn't yet cover — `redundant reference in write! argument` in `crates/zeroclaw-tools/src/claude_code_runner.rs` (lines 211 and 215). `session_id` and `hook_url` are both `String`, so the `&` in the `write!` args is redundant and errors under `-D warnings` on 1.97.

Since this PR is the designated home for clearing 1.97 Clippy warnings, could you fold in the fix here? Suggested patch:

```diff
diff --git a/crates/zeroclaw-tools/src/claude_code_runner.rs b/crates/zeroclaw-tools/src/claude_code_runner.rs
index a31b582975..8ab6c8c7b9 100644
--- a/crates/zeroclaw-tools/src/claude_code_runner.rs
+++ b/crates/zeroclaw-tools/src/claude_code_runner.rs
@@ -208,11 +208,11 @@ impl Tool for ClaudeCodeRunnerTool {
         }
         // Pass session metadata via env vars so the hook can correlate events
         use std::fmt::Write;
-        let _ = write!(env_exports, "CLAUDE_CODE_SESSION_ID={} ", &session_id);
+        let _ = write!(env_exports, "CLAUDE_CODE_SESSION_ID={} ", session_id);
         if let Some(ref ch) = slack_channel {
             let _ = write!(env_exports, "CLAUDE_CODE_SLACK_CHANNEL={} ", ch);
         }
-        let _ = write!(env_exports, "CLAUDE_CODE_HOOK_URL={} ", &hook_url);
+        let _ = write!(env_exports, "CLAUDE_CODE_HOOK_URL={} ", hook_url);
 
         // Create tmux session
         let create_result = Command::new("tmux")
```

Behavior is unchanged (both are `String`, printed via `Display`, and still reused later via `format!`). This would let #9501 be the single sweep for all 1.97 Clippy fixes.

**Comment by Audacity88** [comment]
Thanks for catching these @NiuBlibing  . I folded both redundant `write!` references into `f409c3c68e`; `session_id` and `hook_url` remain available to the later formatting calls. Focused `cargo clippy -p zeroclaw-tools --lib --no-deps -- -D warnings` passes. Required CI completed with all repair-relevant checks passing; the remaining blockers are the newly published `RUSTSEC-2026-0219` advisory against unchanged `nostr 0.44.5` and an unchanged parallel runtime lifecycle test.
